pool/client.go

```diff
@@ -754,11 +754,18 @@ func (c *Client) updateWork(cleanJob bool) {
 	subscribed := c.subscribed
 	c.statusMtx.RUnlock()
 
+	c.mtx.RLock()
+	id := c.id
+	c.mtx.RUnlock()
+
 	if !subscribed || !authorized {
+		log.Tracef("can only timestamp-roll work for subscribed " +
```

Comment:
I think this, and the other new log line, would be acceptable at debug level? I don't think this will be spammed too often?